network/dime_message.ksy

```diff
@@ -0,0 +1,74 @@
+meta:
+  id: dime_message
+  title: DIME (Direct Internet Message Encapsulation) Message
+  xref:
+    wikidata: Q1227457
+  license: CC0-1.0
+  bit-endian: be
+  endian: be
+doc: |
+  Direct Internet Message Encapsulation (DIME)
+  is an old Microsoft specification for sending and receiving
+  SOAP messages along with additional attachments,
+  like binary files, XML fragments, and even other
+  SOAP messages, using standard transport protocols like HTTP.
+doc-ref: 
+  - http://xml.coverpages.org/draft-nielsen-dime-02.txt
+  - https://docs.microsoft.com/en-us/archive/msdn-magazine/2002/december/sending-files-attachments-and-soap-messages-via-dime
+seq:
+  - id: records
+    type: record
+    repeat: eos
+types:
+  padding:
+    doc: padding to the next 4-byte boundary
+    seq:
+      - id: boundary_padding
+        size: (4 - _io.pos) % 4
```

Comment:
Wouldn't this make more sense? It is easier to understand for me because it does not apply modulo to a negative number.

```suggestion
        size: 4 - (_io.pos % 4)
```

This way, if `_io.pos` is `23`, you get `_io.pos % 4 == 3` (because <code>23 == 5 * 4 + <strong>3</strong></code>), but you want to align it to the next 4-byte boundary, which is `24 == 6 * 4`. To achieve that, you need the supplement of the remainder (which is `3`) to 4 and that can be calculated as `4 - 3 == 1`.

IMHO this is more understandable than `(4 - _io.pos) % 4`, because with `_io.pos == 23`, `(4 - _io.pos) == -19` and you're taking a modulo from a negative number here, which is a bit unnatural.

Reply:
The simplification looks good.

I don't think much more can be done, from the logic of the matter it has to be in the format of `(someexpresion) % 4` (the modulo has to be applied last) and to my best knowledge there isn't a simpler or more expressive way than the simplified original one you proposed.